ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionIndexing.scala

```diff
@@ -178,11 +178,9 @@ object TransactionIndexing {
           addEventAndDisclosure(event)
           addVisibility(exercise.targetCoid, blinding.disclosure(nodeId))
           addDivulgence(exercise.targetCoid)
+          addStakeholders(nodeId, exercise.signatories ++ exercise.actingParties)
           if (exercise.consuming) {
-            addStakeholders(nodeId, exercise.stakeholders)
             addArchive(exercise.targetCoid)
-          } else {
-            addStakeholders(nodeId, Set.empty)
           }
```

Comment:
This is the only change we need to make in the way we store exercised events to make the feature work.
We need to store witnesses not only for consuming events as we did until now but also for non-consuming ones. Otherwise querying non-consuming events would not be possible.